docs/Zeitwerk_support/File-structure-and-migration.md

```diff
@@ -0,0 +1,139 @@
+          
+            end
+          end
+          ```
+        - **Exceptions** can be defined in `easy_engines/my_cool_plugin/lib/my_cool_plugin/engine.rb`\
+          This way you can get around module scoping, which we used for these most used models to avoid rewriting
+          everything in `easy_extensions`
+          ```ruby
+          paths['app/models'] << 'app/models/api_services_for_exchange_rates'
+          paths['app/models'] << 'app/models/easy_entity_actions'
+          paths['app/models'] << 'app/models/easy_page_modules'
+          paths['app/models'] << 'app/models/easy_queries'
+          paths['app/models'] << 'app/models/easy_rakes'
+          ```
+- **Initializers**
+    - `easy_extensions` initializers (`easy_engines/easy_extensions/config/initializers/`) have been refactored into
+      many smaller files that serve as a template for **where to put each initialization step**
+    - Default `EasySetting` values are set in `easy_engines/easy_extensions/config/initializers/08_easy_settings.rb`,
+      DON'T use migrations. If you find `EasySetting` being created inside a migration move it to an initializer named
+      `08_easy_settings.rb`(don't be afraid to move other ones down if they already use `08` and higher)
+    - _Stop touching constants_. Remove all constant touching in the initialization phase
+- **Lib**
+  > Many of our plugins are using `lib` as a dumping ground for all sorts of modules, classes and scripts relying
+  on `lib` being autoloaded. That will no longer be the case. `Lib` should only contain files that are loaded manually
+  when needed. The rest should be distributed to other autoloaded directories.
+    - To start, rename the `lib/` directory to `easy_lib/` which will be autoloaded to keep backwards compatibility
+    - Then create a new `lib/` and start moving files from `easy_lib/` to directories where when belong
+      until `easy_lib/` is empty and you can delete it
+    - Don't forget to add explicit requires where needed since `lib` is not autoloaded anymore
+- **Patches**
+    - Similar to _Lib_, start by renaming `patches/` to `easy_patches/` and creating a new `patches/` directory
+    - Then rewrite each patch to use `Rys::Patcher` according to https://github.com/easysoftware/rys/wiki/Patches and
+      move it into patches until `easy_patches/` is empty and you can delete it
+- **Dependencies**
+- **Migrations**
+    - Make sure your migrations are divided to **schema** and **data** migrations and inherit from the respective
+      classes to
+      ensure migration order:
+        - schema migrations in `db/migrate/` directory inheriting from `ActiveRecord::Migration[6.1]`
+        - data migrations in `db/data/` directory inheriting from `EasyExtensions::EasyDataMigration`
+        - **RYSes** have an extra directory `db/after_plugins/` where you should put all **schema** migrations that need
+          to run after all platform engines' migrations
+- **Tests**
+
```

Comment:
missing step how verify that transited plugin is full compatible with zeitwerk. how do I make sure that after removing backward compatibility, my plugin will really work?